File: mobile/lib/features/pets/domain/entities/pet.dart
import 'package:equatable/equatable.dart';

class Pet extends Equatable {
  final String id;
  final String nome;
  final String especie;
  final String raca;
  final int idade;
  final String sexo;
  final String porte;
  final String cor;
  final String? pelagem;
  final String? cidade;
  final String? estado;
  final String imagemUrl;
  final List<String> fotosUrls;
  final String statusAdocao;
  final String? nomeOrganizacao;
  final String? organizacaoId;
  final bool microchipado;
  final bool vacinado;
  final bool castrado;
  final String? adotanteId;

  const Pet({
    required this.id,
    required this.nome,
    required this.especie,
    required this.raca,
    required this.idade,
    required this.sexo,
    required this.porte,
    required this.cor,
    this.pelagem,
    this.cidade,
    this.estado,
    required this.imagemUrl,
    this.fotosUrls = const [],
    required this.statusAdocao,
    this.nomeOrganizacao,
    this.organizacaoId,
    this.microchipado = false,
    this.vacinado = false,
    this.castrado = false,
    this.adotanteId,
  });

  bool get disponivel => statusAdocao == 'DISPONIVEL';

  Pet copyWith({
    String? id,
    String? nome,
    String? especie,
    String? raca,
    int? idade,
    String? sexo,
    String? porte,
    String? cor,
    String? pelagem,
    String? cidade,
    String? estado,
    String? imagemUrl,
    List<String>? fotosUrls,
    String? statusAdocao,
    String? nomeOrganizacao,
    String? organizacaoId,
    bool? microchipado,
    bool? vacinado,
    bool? castrado,
    String? adotanteId,
  }) {
    return Pet(
      id: id ?? this.id,
      nome: nome ?? this.nome,
      especie: especie ?? this.especie,
      raca: raca ?? this.raca,
      idade: idade ?? this.idade,
      sexo: sexo ?? this.sexo,
      porte: porte ?? this.porte,
      cor: cor ?? this.cor,
      pelagem: pelagem ?? this.pelagem,
      cidade: cidade ?? this.cidade,
      estado: estado ?? this.estado,
      imagemUrl: imagemUrl ?? this.imagemUrl,
      fotosUrls: fotosUrls ?? this.fotosUrls,
      statusAdocao: statusAdocao ?? this.statusAdocao,
      nomeOrganizacao: nomeOrganizacao ?? this.nomeOrganizacao,
      organizacaoId: organizacaoId ?? this.organizacaoId,
      microchipado: microchipado ?? this.microchipado,
      vacinado: vacinado ?? this.vacinado,
      castrado: castrado ?? this.castrado,
      adotanteId: adotanteId ?? this.adotanteId,
    );
  }

  @override
  List<Object?> get props => [
        id,
        nome,
        especie,
        raca,
        idade,
        sexo,
        porte,
        cor,
        pelagem,
        cidade,
        estado,
        imagemUrl,
        fotosUrls,
        statusAdocao,
        nomeOrganizacao,
        organizacaoId,
        microchipado,
        vacinado,
        castrado,
        adotanteId,
      ];
}

File: backend/src/main/java/com/Mybuddy/Myb/Model/Pet.java
package com.Mybuddy.Myb.Model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.DocumentReference;
import lombok.*;
import java.util.HashSet;
import java.util.Set;
import java.time.LocalDateTime;


@Document(collection = "pets")
@Getter
@Setter
@NoArgsConstructor
@ToString
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class Pet implements Identifiable {

    @Id
    @EqualsAndHashCode.Include
    private Long id;

    private String nome;

    private String raca;

    private Integer idade;

    private Especie especie;

    private Porte porte;

    private String cor;

    private String pelagem;

    private String sexo;

    @ToString.Exclude
    private Set<FotoPet> fotos = new HashSet<>();

    private StatusAdocao statusAdocao = StatusAdocao.DISPONIVEL;

    @DocumentReference(lazy = true)
    @JsonBackReference
    @ToString.Exclude
    private Organizacao organizacao;

    private boolean microchipado;

    private boolean vacinado;

    private boolean castrado;

    private String cidade;

    private String estado;

    private Long adotanteId;

    private Double peso;

    private String descricao;

    @CreatedDate
    private LocalDateTime dataCriacao;

    @LastModifiedDate
    private LocalDateTime dataAtualizacao;

    public Pet(String nome, String raca, Integer idade, Especie especie, Porte porte,
               String cor, String pelagem, String sexo, Organizacao organizacao,
               boolean microchipado, boolean vacinado, boolean castrado,
               String cidade, String estado) {
        this.nome = nome;
        this.raca = raca;
        this.idade = idade;
        this.especie = especie;
        this.porte = porte;
        this.cor = cor;
        this.pelagem = pelagem;
        this.sexo = sexo;
        this.organizacao = organizacao;
        this.microchipado = microchipado;
        this.vacinado = vacinado;
        this.castrado = castrado;
        this.cidade = cidade;
        this.estado = estado;
        this.statusAdocao = StatusAdocao.DISPONIVEL;
    }

    public void addFoto(FotoPet foto) {
        if (foto != null && !this.fotos.contains(foto)) {
            this.fotos.add(foto);
        }
    }

    public void removeFoto(FotoPet foto) {
        if (foto != null && this.fotos.contains(foto)) {
            this.fotos.remove(foto);
        }
    }

    public void clearFotos() {
        this.fotos.clear();
    }
}